packages/web/src/store/application/ui/buy-audio/sagas.ts

```diff
@@ -1052,6 +1055,7 @@ function* recoverPurchaseIfNecessary() {
   try {
     // Bail if not enabled
     yield* call(waitForWrite)
+    yield* call(waitForAccount)
```

Comment:
`waitForWrite` already calls `waitForAccount`, so we may need something else to detect when the wallet addresses are actually available.
Could do the recovery as a response to the signedIn action being fired instead?

Reply:
Weird! adding this fixed it for me... I wonder if it just delayed it enough and the race condition was less noticeable?

Does the signin action you're proposing happen if the user is returning to a logged in session? I could also `waitForValue` on the wallet? TBH seems like `waitForWrite` should "just work" - maybe it does whatever it needs to do internally to ensure the wallets are present?

Reply:
Updated to `waitForValue` wdyt?

Reply:
`waitForValue` works for me